addons/sale/models/sale_order.py

```diff
@@ -1069,6 +1069,32 @@ def has_to_be_paid(self, include_draft=False):
         transaction = self.get_portal_last_transaction()
         return (self.state == 'sent' or (self.state == 'draft' and include_draft)) and not self.is_expired and self.require_payment and transaction.state != 'done' and self.amount_total
 
+    def _notify_get_recipients_groups(self, msg_vals=None):
+        """ Give access button to users and portal customer as portal is integrated
+        in sale. Customer and portal group have probably no right to see
+        the document so they don't have the access button. """
+        groups = super(SaleOrder, self)._notify_get_recipients_groups(msg_vals=msg_vals)
+        if self._context.get('proforma'):
+            for group in [g for g in groups if g[0] in ('portal_customer', 'customer')]:
+                group[2]['has_button_access'] = False
```

Comment:
Reading this I'm starting to wonder if the result of "_notify_get_recipients_groups" should not be a list of dict instead.
Would be easier to modify it.

But I'm guessing that involves a lot of changes.